src/ipc/ipc4/helper.c

```diff
@@ -382,6 +379,10 @@ int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 		goto e_sink_connect;
 	}
 
+	ret = comp_bind(source, bu);
+	if (ret < 0)
+		goto e_src_bind;
```

Comment:
Error handling in e_src_bind and e_sink_connect implementation needs to be changed when the patch changes the order.

Reply:
I have changed the order